File: data/src/main/java/org/cryptomator/data/cloud/crypto/CryptoConstants.java
package org.cryptomator.data.cloud.crypto;

public class CryptoConstants {

	public static final String MASTERKEY_SCHEME = "masterkeyfile";

	static final String MASTERKEY_FILE_NAME = "masterkey.cryptomator";

	static final String ROOT_DIR_ID = "";
	static final String DATA_DIR_NAME = "d";
	static final String VAULT_FILE_NAME = "vault.cryptomator";
	static final String MASTERKEY_BACKUP_FILE_EXT = ".bkup";

	static final int DEFAULT_MASTERKEY_FILE_VERSION = 999;
	static final int MAX_VAULT_VERSION = 8;
	static final int MAX_VAULT_VERSION_WITHOUT_VAULT_CONFIG = 7;
	static final int VERSION_WITH_NORMALIZED_PASSWORDS = 6;
	static final int MIN_VAULT_VERSION = 5;

	static final int DEFAULT_MAX_FILE_NAME = 220;

	static final byte[] PEPPER = new byte[0];

	static final VaultCipherCombo DEFAULT_CIPHER_COMBO = VaultCipherCombo.SIV_CTRMAC;

}

File: data/src/main/java/org/cryptomator/data/cloud/crypto/CryptoConstants.kt
package org.cryptomator.data.cloud.crypto

import org.cryptomator.cryptolib.api.CryptorProvider

object CryptoConstants {

	const val MASTERKEY_SCHEME = "masterkeyfile"
	const val MASTERKEY_FILE_NAME = "masterkey.cryptomator"
	const val ROOT_DIR_ID = ""
	const val DATA_DIR_NAME = "d"
	const val VAULT_FILE_NAME = "vault.cryptomator"
	const val MASTERKEY_BACKUP_FILE_EXT = ".bkup"
	const val DEFAULT_MASTERKEY_FILE_VERSION = 999
	const val MAX_VAULT_VERSION = 8
	const val MAX_VAULT_VERSION_WITHOUT_VAULT_CONFIG = 7
	const val VERSION_WITH_NORMALIZED_PASSWORDS = 6
	const val MIN_VAULT_VERSION = 5
	const val DEFAULT_MAX_FILE_NAME = 220
	val PEPPER = ByteArray(0)
	val DEFAULT_CIPHER_COMBO = CryptorProvider.Scheme.SIV_CTRMAC

}
